common/cache/lru.go

```diff
@@ -424,3 +444,57 @@ func (c *lru) updateEntryRefCount(entry *entryImpl) {
 		}
 	}
 }
+
+func (c *lru) Stop() {
+	c.loops.Cancel()
+}
+
+func (c *lru) evictLoop(ctx context.Context) error {
+	ch, t := c.timeSource.NewTimer(c.activeExpiry().LoopInterval)
+	for {
+		select {
+		case <-ch:
+			if c.activeExpiry().Enabled {
```

Comment:
tiny optimization: maybe grab the settings once here and pass MaxEntryPerCall into evict?